Heart/User/Domain/Entities/AddressEntity.php

```diff
@@ -5,11 +5,11 @@
 class AddressEntity implements \JsonSerializable
 {
     public function __construct(
-        public readonly string $id,
-        private readonly ?string $country,
-        private readonly ?string $state,
-        private readonly ?string $city,
-        private readonly ?int $zipCode,
+        public string $id,
+        private ?string $country,
+        private ?string $state,
+        private ?string $city,
+        private ?int $zipCode,
     ) {
     }
 
```

Comment:
I think is nice to move the creation logic to the model layer $addressModel->toEntity(). But it's just another suggestion, it's fine as it is right now.